src/main/java/net/countercraft/movecraft/combat/features/BlockBehaviorOverride.java

```diff
@@ -0,0 +1,421 @@
+        final Map<Material, Pair<Integer, Optional<Integer>>> burnOddsMapping = new HashMap<>();
+        final Map<Material, Pair<Integer, Optional<Integer>>> igniteOddsMapping = new HashMap<>();
+
+        // First: Collect them all
+        loadBlastResistanceValues(config, materialList::add, blastResMapping::put);
+        loadFlammabilityValues(config, materialList::add, burnOddsMapping::put, igniteOddsMapping::put);
+
+        // Second: Loop over and create the override objects
+        for(Material m : materialList) {
+            Optional<Float> blastResOverride = Optional.empty();
+            Optional<Integer> burnOddsOverride = Optional.empty();
+            Optional<Integer> burnOddsVanilla = Optional.empty();
+            Optional<Integer> igniteOddsOverride = Optional.empty();
+            Optional<Integer> igniteOddsVanilla = Optional.empty();
+
+            if (blastResMapping.containsKey(m)) {
+                blastResOverride = Optional.ofNullable(blastResMapping.getOrDefault(m, null));
+            }
+
+            if (burnOddsMapping.containsKey(m)) {
+                burnOddsOverride = Optional.ofNullable(burnOddsMapping.get(m).left());
+            }
+
+            if (igniteOddsMapping.containsKey(m)) {
+                igniteOddsOverride = Optional.ofNullable(igniteOddsMapping.get(m).left());
+                igniteOddsVanilla = igniteOddsMapping.get(m).right();
+            }
+
+            BlockOverride override = new BlockOverride(blastResOverride, burnOddsOverride, igniteOddsOverride, burnOddsVanilla, igniteOddsVanilla);
+            BLOCK_OVERRIDES.put(m, override);
+        }
+    }
+
+    protected static void loadFlammabilityValues(FileConfiguration config, Function<Material, Boolean> addToSet, BiConsumer<Material, Pair<Integer, Optional<Integer>>> putBurnOddsFunction, BiConsumer<Material, Pair<Integer, Optional<Integer>>> putIgniteOddsFunction) {
+        if (!config.contains("FlammabilityOverride"))
+            return;
+        var section = config.getConfigurationSection("FlammabilityOverride");
+        if (section == null)
+            return;
+
+        for (var entry : section.getValues(false).entrySet()) {
+            EnumSet<Material> materials = Tags.parseMaterials(entry.getKey());
+            for (Material m : materials) {
+                int burnOddOverride = -1;
+                Optional<Integer> burnOddVanilla = NMS_HELPER.getBurnOdds(m);
+                int igniteOddOverride = -1;
+                Optional<Integer> igniteOddVanilla = NMS_HELPER.getIgniteOdds(m);
+
+                String valStr = entry.getValue().toString();
+                try {
+                    String[] split = valStr.split(",");
+                    burnOddOverride = Integer.parseInt(split[0]);
+                    igniteOddOverride = burnOddOverride;
+                    if (split.length > 1) {
+                        igniteOddOverride = Integer.parseInt(split[1]);
```

Comment:
`RequiredBlockEntry` is probably a good reference for loading a list of two integers safely.  `TypeData.java`:
```java
    /**
     * Gets the requested Set of <code>RequiredBlockEntry</code>s by its key.
     * If the key is not found, an error is thrown.
     * If the value is found, but is not a Set of <code>RequiredBlockEntry</code>s, an error is thrown.
     *
     * @param key - Key of the Set of <code>RequiredBlockEntry</code>s to get
     * @return The requested Set of <code>RequiredBlockEntry</code>s
     */
    @NotNull
    public Set<RequiredBlockEntry> getRequiredBlockEntrySet(@NotNull String key) {
        var data = getData(key).getBackingData();
        Set<RequiredBlockEntry> out = new HashSet<>();
        for(var entry : data.entrySet()) {
            EnumSet<Material> materials = parseMaterials(key, entry.getKey());

            var limits = (ArrayList<?>) entry.getValue();
            if(limits.size() != 2)
                throw new IllegalArgumentException("Block entry range for key " + key + " and value '" + entry.getKey()
                        + "' must be a pair, but found " + limits.size() + " entries");
            var min = parseLimit(limits.get(0));
            var max = parseLimit(limits.get(1));

            String name;
            String s = null;
            Object entryKey = entry.getKey();
            if (entryKey instanceof ArrayList) {
                //name for array list
                ArrayList<?> list = (ArrayList<?>) entryKey;
                if (list.get(0) instanceof String) {
                    s = (String)list.get(0);
                }
            } else if (entryKey instanceof String) {
                s = (String)entryKey;
            }
            if (s == null) {
                out.add(new RequiredBlockEntry(materials, min, max, null));
                continue;
            }
            if (s.charAt(0) == '#' && s.length() > 1) {
                int n = 1;
                if (s.contains(":")) {
                    s = s.split(":")[1];
                    n--;
                }
                name = s.substring(n);
            } else {
                name = s;
            }
            out.add(new RequiredBlockEntry(materials, min, max, name));
        }
        return out;
    }
```

Which loads YAML of the following format: `Airship.craft`
```yml
flyblocks:
    "#wool": #wool represents lift
        - 50.0
        - 100.0
    "#chests": # limit chests to make big airships important for merchant missions
        - 0.0
        - 1.0
    [dispenser,hopper,dropper]: # dispensers and hoppers can be used for inventory too, so limit them as well
        - 0.0
        - 2.0
    furnace: # Furnaces for cargo? really guys?
        - 0.0
        - N50
```

You won't need the piece to calculate out a name for the materials, but parsing the YAML should be similar.  Instead of listing the two integers with extra lines and `-`s, you can also format them this way:

```yml
flyblocks:
    "#wool": [50.0, 100.0]
```

Reply:
Resolved it a bit differently, it's something like this now:
`
FlammabilityOverride:
  obsidian:
    burnOddity: 20
    igniteOddity: 100
`

I think that is cleaner and easier to read, lmk if that is alright too